pyexasol/callback.py

```diff
@@ -74,33 +78,88 @@ def import_from_pandas(pipe, src, **kwargs):
-            index=False,
-            lineterminator="\n",
-            quoting=csv.QUOTE_NONNUMERIC,
-            **kwargs
-        )
-    else:
-        return src.to_csv(
-            wrapped_pipe,
-            header=False,
-            index=False,
-            line_terminator="\n",
-            quoting=csv.QUOTE_NONNUMERIC,
-            **kwargs
+
+def get_parquet_files(source: Union[list[Path], Path, str]) -> list[Path]:
+    if isinstance(source, str):
+        matches = glob.glob(source)
+        return sorted(
+            [filepath for i in matches if (filepath := Path(i)) and filepath.is_file()]
         )
+    if isinstance(source, Path):
+        if source.is_file():
+            return [source]
+        elif source.is_dir():
+            return sorted(source.glob("*.parquet"))
+    if isinstance(source, list):
+        not_a_valid_path_file = [
+            filepath
+            for filepath in source
+            if not isinstance(filepath, Path) or not filepath.is_file()
+        ]
+        if len(not_a_valid_path_file) > 0:
+            raise ValueError(
+                f"source {source} contained entries which were not `Path` or valid `Path` files {not_a_valid_path_file}"
+            )
+        return source
+    raise ValueError(
+        f"source {source} is not a supported type (Union[list[Path], Path, str])."
+    )
+
+
+def import_from_parquet(pipe, source: Union[list[Path], Path, str], **kwargs):
+    """
+    Basic example how to import from pyarrow parquet file(s)
+
+    Args:
+        source: Local filepath specification(s) to process. Can be one of:
+            - list[pathlib.Path]: list of specific files
+            - pathlib.Path: can be either a file or directory. If it's a directory,
+            all files matching this pattern *.parquet will be processed.
+            - str: representing a filepath which already contains a glob pattern
+            (e.g., "/local_dir/*.parquet")
+        **kwargs:
+            Custom params for "pyarrow.csv.WriteOptions"
+
+    Please note that nested or hierarchical types are not supported.
+    """
+    from pyarrow import (
+        csv,
+        parquet,
+        types,
+    )
+
+    if not (parquet_files := get_parquet_files(source)):
+        raise ValueError(f"source {source} does not match any files")
+
+    for file in parquet_files:
+        parquet_file = parquet.ParquetFile(file)
+        num_row_groups = parquet_file.num_row_groups
+
+        for i in range(num_row_groups):
+            row_group_table = parquet_file.read_row_group(i)
+            for field in row_group_table.schema:
+                if types.is_nested(field.type):
+                    raise ValueError(
+                        f"Field {field} of schema from file {file} is hierarchical which is not supported."
+                    )
+
+            write_options = csv.WriteOptions(include_header=False, **kwargs)
+            csv.write_csv(row_group_table, pipe, write_options=write_options)
```

Comment:
Also the order in which we write the columns, might be something people want to customize.

Reply:
Is it ok for the order of the columns to come from the iter_batches?